### PR TITLE
[tests-only][full-ci] bug demo test for removing files_trashbin app from whitelist for guest user

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR adds test for bug demonstration for adding and removing `files_trashbin` from whilelist for guest user, to check if the guest user are able to delete the shared files and if the deleted files are available in the trashbin.

## Related Issue
Related to issue: https://github.com/owncloud/guests/issues/553

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test added

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/apiGuests/guests.feature [hunk 1]
```diff
@@ -478,3 +478,48 @@ Feature: Guests
     Then the HTTP status code should be "204"
     # uncomment the line above and remove the line above when issue-549 is fixed
```
Comment (grgprarup) [inline tests/acceptance/features/apiGuests/guests.feature:479]:
```suggestion
    # uncomment the line below and remove the line above when issue-549 is fixed
```

File: tests/acceptance/features/apiGuests/guests.feature [hunk 2]
```diff
@@ -478,3 +478,48 @@ Feature: Guests
...
+    And user "Alice" has shared folder "/tmp" with user "guest@example.com"
+    And guest user "guest" has registered
+    And the administrator has limited the guest access to the default whitelist apps
+    And the administrator has added the app "files_trashbin" to the whitelist for the guest user
+    When user "guest@example.com" deletes file "/tmp/textfile0.txt" using the WebDAV API
+    Then the HTTP status code should be "204"
+    And as "Alice" file "/tmp/textfile0.txt" should not exist
+    When user "Alice" tries to list the trashbin content for user "Alice"
+    Then the HTTP status code should be "207"
+    And as "Alice" the file with original path "/tmp/textfile0.txt" should exist in the trashbin
```
Comment (grgprarup) [inline tests/acceptance/features/apiGuests/guests.feature:497]: Do we need this test?

File: tests/acceptance/features/apiGuests/guests.feature [hunk 3]
```diff
@@ -478,3 +478,48 @@ Feature: Guests
...
+    Given user "Alice" has been created with default attributes and small skeleton files
+    And the administrator has created guest user "guest" with email "guest@example.com"
+    And user "Alice" has uploaded file with content "some content" to "textfile0.txt"
+    And user "Alice" has shared file "/textfile0.txt" with user "guest@example.com"
+    And guest user "guest" has registered
+    And the administrator has limited the guest access to the default whitelist apps
+    And the administrator has removed the app "files_trashbin" from the whitelist for the guest user
+    When user "guest@example.com" deletes file "/textfile0.txt" using the WebDAV API
+    Then the HTTP status code should be "204"
+    # uncomment the line below and remove the line above when issue-553 is fixed
```
Comment (grgprarup) [inline tests/acceptance/features/apiGuests/guests.feature:524]: Can't the user delete a file if the files_trashbin app is not whitelisted? I think the user should allow deleting files, but cannot access/delete files in the trashbin.

File: tests/acceptance/features/apiGuests/guests.feature [hunk 4]
```diff
@@ -478,3 +478,48 @@ Feature: Guests
     Then the HTTP status code should be "204"
     # uncomment the line above and remove the line above when issue-549 is fixed
     #Then the HTTP status code should be "403"
+
+  @email
+  Scenario: files inside shared folder deleted by guest user is available in sharer trashbin when files_trashbin app is whitelisted
```
Comment (grgprarup) [inline tests/acceptance/features/apiGuests/guests.feature:483]: In this scenario, the folder is shared but the file inside the folder is deleted and in scenario L#500 the shared file is deleted so what's the difference between the two scenarios?

File: tests/acceptance/features/apiGuests/guests.feature (hunk 3 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiGuests/guests.feature:524]: When the files_trashbin app is not whitelisted the in the UI of the guest, the delete option is not available and the guest users are not able to delete files.

File: tests/acceptance/features/apiGuests/guests.feature (hunk 4 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiGuests/guests.feature:483]: The actual behavior of the two scenarios are inconsistent. In this scenario the file gets deleted on the sharer side while in the scenario L500 the file is not deleted on the sharer side. It is discussed in https://github.com/owncloud/guests/issues/553#issuecomment-1469264993
I missed to put the issue tag.

File: tests/acceptance/features/apiGuests/guests.feature [hunk 5]
```diff
@@ -476,5 +476,50 @@ Feature: Guests
...
+    Given user "Alice" has been created with default attributes and small skeleton files
+    And the administrator has created guest user "guest" with email "guest@example.com"
+    And user "Alice" has uploaded file with content "some content" to "textfile0.txt"
+    And user "Alice" has shared file "/textfile0.txt" with user "guest@example.com"
+    And guest user "guest" has registered
+    And the administrator has limited the guest access to the default whitelist apps
+    And the administrator has removed the app "files_trashbin" from the whitelist for the guest user
+    When user "guest@example.com" deletes file "/textfile0.txt" using the WebDAV API
+    Then the HTTP status code should be "204"
+    # uncomment the line below and remove the line above when issue-553 is fixed
```
Comment (phil-davis) [inline tests/acceptance/features/apiGuests/guests.feature:524]: The behavior is actually correct. The server will "unshare"  the file, and trashbin code will not need to be used. You can test the behavior using the same steps as in the previous scenario:
```
And as "Alice" file "/textfile0.txt" should exist
And as "guest@example.com" file "/textfile0.txt" should not exist
```

**Review by phil-davis — APPROVED** [review]

**Pull request merged**